### PR TITLE
docs: added events documentation

### PR DESCRIPTION
**Pull request by shahednasser**
- Updated the events overview and create subscriber guides for better structure and with new information
- Added the Create Event Module guide.
- Added guides on how to install the redis and local modules

This PR depends on #3602 as it refers to the modules documentation pages.

Closes DOCS-639, DOCS-654, DOCS-655, DOCS-656, and DOCS-673

### CODE REVIEW

File: docs/content/development/events/create-module.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+addHowToData: true
+---
+
+# Create Events Module
+
+In this document, you’ll learn how to create an events module.
+
+## Overview
+
+Medusa provides ready-made modules for events, including the local and Redis modules. If you prefer another technology used for managing events, you can build a module and publish it as an NPM package.
```
[the remark this answers is not in the document]
Comment (olivermrbl) [reply docs/content/development/events/create-module.md:12]: My suggestion here is to remove the "publish it as an npm package" and instead add a small guide (a different place) on how to use custom modules in your project - and then link to it from all our "Create "X" module" guides. 

Reason being, that you can build and use modules in two ways; add it directly to your project and reference the file path in your `medusa-config.js` or publish it as an NPM package. 

I am fine to add both options here, but I thought a general guide would be more reasonable as we could use it elsewhere.

File: docs/content/development/events/modules/redis.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+```
+
+---
+
+## Step 2: Add Environment Variable
+
+The Redis events module requires a connection URL to Redis as part of its options. If you don’t already have an environment variable set for a Redis URL, make sure to add one:
+
+```bash
+CACHE_REDIS_URL=<YOUR_REDIS_URL>
```
Comment (olivermrbl) [inline docs/content/development/events/modules/redis.md:47]:
```suggestion
REDIS_URL=<YOUR_REDIS_URL>
```

File: docs/content/development/modules/create.mdx [hunk 3]
```diff
@@ -0,0 +1,321 @@
+---
```
Comment (olivermrbl) [inline docs/content/development/modules/create.mdx:1]: comment: I will move to the other PR to review this :)

**Review by olivermrbl — CHANGES_REQUESTED** [review]
This is really a great piece of work @shahednasser 💯 

Have added a couple of comments and todos. I will move to the other PR for things related to Create a Module documentation.

File: docs/content/development/events/create-module.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shahednasser) [reply docs/content/development/events/create-module.md]: I think this makes a lot of sense! I was wondering about that too, but couldn't find anything to support my thinking.

I added the two possibilities: referencing and publishing. I think we should highlight both approaches (but we should add the how in the Create Module guide).

File: docs/content/development/events/create-module.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    this.moduleOptions = options
+  }
+
+  // ...
+}
+```
+
+### emit
+
+The `emit` method is used to push an event into the queue to be triggered, which allows subscribers to that event to execute their asynchronous tasks.
```
[the remark this answers is not in the document]
Comment (olivermrbl) [reply docs/content/development/events/create-module.md]: Suggested this as the queue is too specific - not applicable to all messaging systems.

[the remark this answers is not in the document]
Comment (shahednasser) [reply docs/content/development/events/create-module.md]: I thought I added this in 😅 will add it in again

**Comment by olivermrbl** [comment]
LGTM 🚢

**Pull request merged**